subsys/bluetooth/host/l2cap.c

```diff
@@ -992,6 +992,11 @@ static uint16_t l2cap_chan_accept(struct bt_conn *conn,
 		return le_err_to_result(err);
 	}
 
+	if (!(*chan)->ops->recv) {
+		BT_ERR("Mandatory callback 'recv' missing");
+		return BT_L2CAP_LE_ERR_UNACCEPT_PARAMS;
+	}
+
```

Comment:
I think we might as well started making it a habit to use boolean values for boolean expressions, as that is likely something we will need to do once the MISRA rules will start being checked proper, i.e. in this case do `if (pointer == NULL)` instead of `if (!pointer)`. 

```suggestion
	if ((*chan)->ops->recv == NULL) {
		BT_ERR("Mandatory callback 'recv' missing");
		return BT_L2CAP_LE_ERR_UNACCEPT_PARAMS;
	}

```

Reply:
Let's do that as part of an overall MISRA PR then. There would be a lot of places where that would be needed.